File: conformance/tests/if-without-else/if-without-else.runar.rs
use runar::prelude::*;

#[runar::contract]
struct IfWithoutElse {
    #[readonly]
    threshold: Int,
}

#[runar::methods(IfWithoutElse)]
impl IfWithoutElse {
    #[public]
    fn check(&self, a: Int, b: Int) {
        let mut count: Int = 0;
        if a > self.threshold {
            count = count + 1;
        }
        if b > self.threshold {
            count = count + 1;
        }
        assert!(count > 0);
    }
}

File: conformance/tests/if-without-else/if-without-else.runar.go
//go:build ignore

package contract

import "runar"

type IfWithoutElse struct {
	runar.SmartContract
	Threshold runar.Int `runar:"readonly"`
}

func (c *IfWithoutElse) Check(a runar.Int, b runar.Int) {
	count := runar.Int(0)
	if a > c.Threshold {
		count = count + 1
	}
	if b > c.Threshold {
		count = count + 1
	}
	runar.Assert(count > 0)
}
